### ISSUE TITLE
A few questions during training of CNN1

### ISSUE THREAD

**Issue by kserrano109** [SchornacklabSLCU/amfinder#17]
Hello,

I am currently working to train a fluorescently stained image set. I have a total of 48 images, which I was able to input into the training pipeline. However, I only ended up with 1500 M+ tiles, 1300 M- tiles, and 1400 X tiles. This resulted in repetitive pre-mature termination at epoch 12-20, which I believe is due to pre-termination built into the code to prevent overfitting if the loss doesn't decrease for 12 training cycles in a row. 

From the publication, these numbers seem very low even though I'm using more images in the training, it seems that there were 175105 total tiles from 32 images. I don't quite understand why how 48 images resulted in dramatically less tiles than 32, but maybe I am missing something big here. 

So my questions are:
1.  Is there a way to increase the tile # from an image? Currently, the max number of tiles I am getting from 1 image is 125 tiles (i.e. my images are only ever divided into 125 grids by the amfbrowser whenever I feed an image to it.)
2. Is the early pre-termination resulting from a small sample size or too few tiles?

Thank you in advance and please let me know if I can clarify on any of this,

Karl

**Comment by EEvangelisti** [comment]
Hello Karl,

Thanks for your interest in AMFinder. Wow, using it with fluorescent images is something I have never tried! I hope you will get relevant results and be able to share feedback once everything is running. Regarding your questions:

1. By default, many background tiles are discarded to equilibrate the classes, hence the low total number of tiles. You can disable this behaviour by providing the option `-k`/`--keep_background` to `amf`, but that won't necessarily improve the final result. You can also alter tile size with option `-t`/`--tile_size` (for example, you could try `-t 50`). The same option works with `amf` and `amfbrowser`. You can provide images with different tile sizes for the training.

2. Pre-termination occurs when loss does not decrease any more. It is indeed often associated with a small number of tiles. For the manuscript, we used large scans containing thousands of tiles each. Given that you have a limited number of images available, I would recommend reducing the tile size and possibly activating data augmentation with option `-a`/`--data_augmentation`. Also, do not worry too much about epochs, in the end, accuracy matters.

Please let me know if you need more information. I also recommend you have a look at the `README` file. It provides details of all the options I mentioned.

**Comment by EEvangelisti** [comment]
I am closing this due to inactivity.
